js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -751,23 +700,14 @@ const EditingController = modules.ViewController.inherit((function() {
         },
 
         getEditFormRowIndex: function() {
-            const editMode = getEditMode(this);
-
-            return editMode === EDIT_MODE_FORM || editMode === EDIT_MODE_POPUP ? this._getVisibleEditRowIndex() : -1;
+            return -1;
```

Comment:
It seems that getEditFormRowIndex should not be in ui.grid_core.editing.js. The same is related to other form and popup methods. From what I gather ui.grid_core.editing is assumed to be a base editing class. Should it know about popup and form?

Reply:
As I think, base editing _can_ know about declarations and contain dummy implementations. This method can be called when mode is not form/popup, so returning -1 in the base class is OK. 
However, when it comes to some more complicated logic (ex. calling _getVisibleEditRowIndex), this should be moved to the formBased. In this pull request I did not move all code related to form/popup edit modes, only easy to move and the most obvious parts.

Reply:
OK.